atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/timestamp/InDbTimestampBoundStore.java

```diff
@@ -45,15 +50,23 @@
     private static final Logger log = LoggerFactory.getLogger(InDbTimestampBoundStore.class);
 
     private final ConnectionManager connManager;
+    private final TableReference timestampTable;
 
     @GuardedBy("this") // lazy init to avoid db connections in constructors
     private DBType dbType;
 
     @GuardedBy("this")
     private Long currentLimit = null;
 
-    public InDbTimestampBoundStore(ConnectionManager connManager) {
+    public static InDbTimestampBoundStore create(ConnectionConfig connConfig) {
+        return new InDbTimestampBoundStore(
+                new HikariCPConnectionManager(connConfig, Visitors.emptyVisitor()),
+                AtlasDbConstants.TIMESTAMP_TABLE);
+    }
+
+    public InDbTimestampBoundStore(ConnectionManager connManager, TableReference timestampTable) {
         this.connManager = Preconditions.checkNotNull(connManager);
+        this.timestampTable = Preconditions.checkNotNull(timestampTable);
```

Comment:
This is all a mess.  There shouldn't be a dependency here and instead this should own the table.  The issue is that internal code physically cannot upgrade around this until later this quarter.  Until then this is necessary.